tests/test.cpp

```diff
@@ -1,42 +1,167 @@
-#include "../Jsonparser.cpp"
+#include "Jsonparser.h"
+#include "Szorny.h"
+#include "Kalandor.h"
 #include "gtest/gtest.h"
 
 
 using namespace std;
 
-
-TEST(Jsontest, Fajlvaltozo) {
+TEST(Jsonparsertest, Fajlvaltozo) {
 	std::ifstream f("Hosarkany.json");
 	Jsonparser beolvasas(f);
 	EXPECT_EQ(beolvasas.getErtek("name"), "Hosarkany");
 	EXPECT_EQ(beolvasas.getErtek("hp"), "300");
 	EXPECT_EQ(beolvasas.getErtek("dmg"), "30");
+	EXPECT_EQ(beolvasas.getErtek("speed"), "2.4");
 }
 
-TEST(Jsontest, Szoveg) {
+TEST(Jsonparsertest, Szoveg) {
 	std::string szoveg = "Sotetvarazslo.json";
 	Jsonparser beolvasas(szoveg);
 	EXPECT_EQ(beolvasas.getErtek("name"), "Sotetvarazslo");
 	EXPECT_EQ(beolvasas.getErtek("hp"), "250");
 	EXPECT_EQ(beolvasas.getErtek("dmg"), "40");
+	EXPECT_EQ(beolvasas.getErtek("speed"), "2.0");
 }
 
 
-TEST(Jsontest, Fajlnev) {
+TEST(Jsonparsertest, Fajlnev) {
 	const char * fajlnev = "Arnykiraly.json";
 	Jsonparser beolvasas(fajlnev);
 	EXPECT_EQ(beolvasas.getErtek("name"), "Arnykiraly");
 	EXPECT_EQ(beolvasas.getErtek("hp"), "200");
 	EXPECT_EQ(beolvasas.getErtek("dmg"), "48");
+	EXPECT_EQ(beolvasas.getErtek("speed"), "2.3");
+}
+TEST(Jsontest, Whitespace_test) {
+	std::ifstream f("Atoksarkany.json");
+	Jsonparser beolvasas(f);
+	EXPECT_EQ(beolvasas.getErtek("name"), "Atoksarkany");
+	EXPECT_EQ(beolvasas.getErtek("hp"), "260");
+	EXPECT_EQ(beolvasas.getErtek("dmg"), "45");
+	EXPECT_EQ(beolvasas.getErtek("speed"), "1.7");
 }
 
+TEST(Jsontest, Kulcssorrend_test) {
+	std::ifstream f("Kardvadasz.json");
+	Jsonparser beolvasas(f);
+	EXPECT_EQ(beolvasas.getErtek("name"), "Kardvadasz");
+	EXPECT_EQ(beolvasas.getErtek("hp"), "280");
+	EXPECT_EQ(beolvasas.getErtek("dmg"), "48");
+	EXPECT_EQ(beolvasas.getErtek("speed"), "2.1");
+}
 
+TEST(Jsontest, Idezojelfuggetlenseg_test) {
+	std::ifstream f("Arnykiraly.json");
+	Jsonparser beolvasas(f);
+	EXPECT_EQ(beolvasas.getErtek("name"), "Arnykiraly");
+	EXPECT_EQ(beolvasas.getErtek("hp"), "200");
+	EXPECT_EQ(beolvasas.getErtek("dmg"), "48");
+	EXPECT_EQ(beolvasas.getErtek("speed"), "2.3");
+}
 
+TEST(Jsontest, Tobb_kevesebb_idozejel_test) {
+	std::ifstream f("Sotetvarazslo.json");
+	Jsonparser beolvasas(f);
+	EXPECT_EQ(beolvasas.getErtek("name"), "Sotetvarazslo");
+	EXPECT_EQ(beolvasas.getErtek("hp"), "250");
+	EXPECT_EQ(beolvasas.getErtek("dmg"), "40");
+	EXPECT_EQ(beolvasas.getErtek("speed"), "2.0");
+}
+TEST(Maintest, Nem_letezo_fajl_test) {
+	std::ifstream f("Lathatatlan.json");
+	try {
+		Jsonparser f1(f);
+	}
+	catch (int e) {
+		EXPECT_EQ(e, MYFILEERROR);
+	}
+}
 
 
+TEST(Maintest, Tobb_kevesebb_adat_test) {
+	std::ifstream f("Exodia.json");
+	try {
+		Jsonparser f1(f);
+	}
+	catch (int e) {
+		EXPECT_EQ(e, INVALID_VALUE);
+	}
+}
 
+TEST(Szornytest, Lko_test) {
+	EXPECT_EQ(round(gcd(2.4,1.6)),round(0.8));
+	EXPECT_EQ(round(gcd(2.2, 4.4)), round(2.2));
+	EXPECT_EQ(round(gcd(1.3, 1.1)), round(0.1));
+}
```

Comment:
a működést értem, de ez hogyan teszteli a játékot?

Reply:
erre a függvényre az attackspeed branchnél volt szükség(saját implementálás), ez nem az osztály függvénye,csak ott lett kifejtve. Azért gondoltam letesztelni,mert igazából a legnagyobb közös osztó lesz a lépésköz az időzítőnél(while ciklusnál), ami meghatározza mikor visz be egy szörny sebzést,mert így biztos, hogy a ciklus elhalad a megfelelő értékek mellett. Ha ez nem működik jól, nem működik a program.

Reply:
oh így érthető, akkor rendben